wordpress-mobile/WordPress-Android
# Media upload error codes self-hosted in media lib

This PR is part of a series related to #12325

The changes here should address the Media Library with self-hosted scenario. The equivalent scenarios for GB Editor will be covered in subsequent PRs.

This has a companion FluxC PR [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1658)

The Fluxc modifications map the 401 and 500 upload media errors that can be found [here](https://github.com/WordPress/WordPress/blob/649ab58d96491549f8b617d2702bb437488bb16d/wp-includes/class-wp-xmlrpc-server.php#L6298)

NOTE: I was trying to use a similar approach as we did in #12620 ; that is leveraging the user error message that the API is actually able to provide and display them to the user. Unfortunately AFAIU in the case of XML-RPC API we do not send or use a `locale`parameter that the client app can send to the server to declare it's idiom. The only thing I was able to get is a localized message changing the language server side, but that is not a per user triggered localization so we cannot use this approach. I reverted to using general string resources that only map the category of the error (401 and 500).

### To test
Access to a pure self-hosted site in the app.

- Open the Media Library screen and try to upload a file
- Force trigger the errors as per the link above (I can provide access to a test site I have setup for that or you can use Charles proxy rewrite feature or AS debugger for this)
- Check you get the relevant string resources message in the snackbar

NOTE: if you are using the Charles proxy approach, the message received from the server should have a 200 http status and the following xml content:

#### 401 case
```
<?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value><int>401</int></value>
        </member>
        <member>
          <name>faultString</name>
          <value><string>Sorry, you are not allowed to upload files.</string></value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>
```

#### 500 case
```
<?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value><int>500</int></value>
        </member>
        <member>
          <name>faultString</name>
          <value><string>Could not write file.</string></value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>
```

PR submission checklist:
